lib/types/json.js

```diff
@@ -80,13 +81,13 @@ function json (options) {
 
       if (first !== '{' && first !== '[') {
         debug('strict violation')
-        throw createStrictSyntaxError(body, first)
+        throw createStrictSyntaxError(parser, body, first)
       }
     }
 
     try {
       debug('parse json')
-      return JSON.parse(body, reviver)
+      return parser(body, reviver)
```

Comment:
I'm not sure this is right for custom parsers. Or at least it's not clear in the docs for the parser option that this is going to happen and how people should build their parser dunxtion to compensate.

Reply:
The other option would be to remove the `reviver` param and rely on people using that parameter should change to this. I didn't do that as that would break BC, and I figure a project like this wouldn't want to break BC very often.

Maybe the right way is to make the docs more clear about this?

Reply:
Sorry, I commented on the wrog line.

Reply:
Actually, I don't think I can agree with you that the docs are not clear: 

> The reviver option is passed directly to the `parser` as the second argument.

If you can suggest an update to that text I'm all ears.

Reply:
I moved my comment down. The new comment for this line is to clarify in the docs the arguments the custom parser will be called with.

Reply:
Remove the line from the reviver docs and add in the specific signature into the parser docs. The current parser section doesn't help someone build  their own, I think it assumes someone is going to understand JSON.parse and copy that? If so, just write that down.